tiledb/sm/array_schema/domain.h

```diff
@@ -575,14 +597,26 @@ class Domain {
   /** The tile order of the array the domain belongs to. */
   Layout tile_order_;
 
+  /**
+   * Vector of functions, one per dimension, for comparing the cell order of
+   * two coordinates. The inputs to the function are:
+   *
+   * - dim: The dimension to check the coordinates on.
+   * - buff: The buffer that stores all coorinates;
+   * - a, b: The positions of the two coordinates in the buffer to compare.
+   */
+  std::vector<int (*)(
+      const Dimension* dim, const QueryBuffer* buff, uint64_t a, uint64_t b)>
+      cell_order_cmp_func_;
+
   /**
    * Vector of functions, one per dimension, for comparing the cell order of
    * two coordinates. The inputs to the function are:
    *
    * - coord_a, coord_b: The two coordinates to compare.
    */
   std::vector<int (*)(const void* coord_a, const void* coord_b)>
-      cell_order_cmp_func_;
+      cell_order_cmp_func_2_;
```

Comment:
Notice the different inputs. One operates on coordinate positions in a buffer, the other on coordinates directly. The only reason why the name is different is because this is a member attribute (vs. a function that can be simply overloaded).